packages/peregrine/lib/store/middleware/auth.js

```diff
@@ -0,0 +1,60 @@
+import BrowserPersistence from '../../util/simplePersistence';
+import userActions, { signOut } from '../actions/user';
+
+const timers = new Map();
+const storage = new BrowserPersistence();
+const SET_TOKEN = userActions.setToken.toString();
+const CLEAR_TOKEN = userActions.clearToken.toString();
+const GET_DETAILS = userActions.getDetails.request.toString();
+
+const isSigningIn = type => type === SET_TOKEN || type === GET_DETAILS;
```

Comment:
Why `|| GET_DETAILS`? Is this a safeguard against tokens invalidating between the token set and the details fetch?

Reply:
When a user starts up the application, if they have a valid token in storage, we sign them in. Since they already have a token, a `SET_TOKEN` action is never dispatched. Instead, the first user action is `GET_DETAILS`.

Reply:
So it's really "is signing in _or_ is getting user details". And in both cases we set a timer based on the time remaining on the TTL. Got it.